src/main/java/net/pms/network/UPNPHelper.java

```diff
@@ -477,12 +477,18 @@ public void run() {
 
 							boolean redundant = address.equals(lastAddress) && packetType == lastPacketType;
 
-							if (packetType == M_SEARCH) {
+							if (packetType == M_SEARCH || packetType == NOTIFY) {
 								if (configuration.getIpFiltering().allowed(address)) {
 									String remoteAddr = address.getHostAddress();
 									int remotePort = receivePacket.getPort();
 									if (!redundant && LOGGER.isTraceEnabled()) {
-										LOGGER.trace("Received a M-SEARCH from [{}:{}]: {}", remoteAddr, remotePort, s);
+										String requestType = "unreognized";
+										if (packetType == M_SEARCH) {
+											requestType = "M-SEARCH";
+										} else if (packetType == NOTIFY) {
+											requestType = "NOTIFY";
+										}
+										LOGGER.trace("Received a " + requestType + " from [{}:{}]: {}", remoteAddr, remotePort, s);
```

Comment:
Why would you use both `+ type +` and `{}` in the same log entry? It means that two `StringBuilder`s have to be created when running this one line in trace mode, and one when not in trace.